components/runtime/squeak5c/Makefile

```diff
@@ -0,0 +1,97 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"). You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2020 David Stes
+#
+
+
+#
+# Package Makefile for the "OpenSmalltalk Cog Spur" Squeak Smalltalk V5 system
+# See http://squeak.org and http://opensmalltalk.org
+#
+
+# opensmalltalk-vm can be built both in 32 and 64bit
+# but generally SqueakV5 images are 64bit
+BUILD_BITS=64
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		squeak-5c
+COMPONENT_VERSION=	5.0.7
+GIT_TAG=		sun-v$(COMPONENT_VERSION)
+PLUGIN_REV=		5.0-202005021055-cog
+COMPONENT_SUMMARY=	The Cog Implementation of the Squeak V5 Smalltalk Virtual Machine
+COMPONENT_PROJECT_URL=	http://www.squeak.org
+COMPONENT_FMRI=		runtime/squeak-5c
+COMPONENT_CLASSIFICATION=	Development/Other Languages
+
+# See http://wiki.squeak.org/squeak/933
+# See http://wiki.squeak.org/squeak/159
+# there's 2 license lines in the manifest, the MIT squeak.license and this one
+COMPONENT_LICENSE=	Squeak5
+COMPONENT_LICENSE_FILE=	squeak5.license
+
+COMPONENT_SRC=		opensmalltalk-vm-$(GIT_TAG)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_HASH=	sha256:50c180b174048acd98a7d9705f0a40a3dc8bdc62b42d88eb7e20e71834184e4c
+COMPONENT_ARCHIVE_URL=	https://codeload.github.com/cstes/opensmalltalk-vm/tar.gz/$(GIT_TAG)
+
+TEST_TARGET= $(NO_TESTS)
+
+include $(WS_MAKE_RULES)/common.mk
+
+PKG_OPTIONS += -D PLUGIN_REV="$(PLUGIN_REV)"
+
+PATH=$(PATH.gnu)
+
+# opensmalltalk configure script checks for plugins files in builddir
+COMPONENT_PRE_CONFIGURE_ACTION = ( cp plugins.ext plugins.int $(BUILD_DIR_64) )
+
+# the default CFLAGS.gcc will be -m64 -O3 but this does not work for us
+# opensmalltalk code is not compatible with the gcc optimizer -O2 or higher
+# also the -g (debug) flag is intentional, perhaps omit-frame-pointer issues
+gcc_OPT=		-g -DNDEBUG -DDEBUGVM=0 -DCOGMTVM=0
+
+CONFIGURE_SCRIPT= $(SOURCE_DIR)/platforms/unix/config/configure
+CONFIGURE_OPTIONS +=	--with-vmversion=5.0
+CONFIGURE_OPTIONS +=	--disable-dynamicopenssl
+CONFIGURE_OPTIONS +=	--without-libtls
+CONFIGURE_OPTIONS +=	--with-src=spur64src
+
+CONFIGURE_ENV +=	TARGET_ARCH="-m64"
+
+# the opensmalltalk Makefile does not use DESTDIR
+COMPONENT_INSTALL_ARGS=  	ROOT=$(PROTO_DIR)
+COMPONENT_INSTALL_TARGETS= 	install-squeak install-doc install-plugins
+
+# Makefile has no strip target
+COMPONENT_POST_INSTALL_ACTION = \
+( $(STRIP) $(PROTOUSRLIBDIR64)/squeak/$(PLUGIN_REV)/squeak )
+
```

Comment:
Stripping is a contradiction to your debug flag (-g)! as stripping removes (among other things) the debug information!
Exerpt from man strip:
strip - strip symbol table, debugging and line number information from
       an object file.

Reply:
There is no contradiction, as the man page explains it 's perfectly normal to strip debugging and line number information from an object file.  The 'strip' command is not a contradiction.   Somebody can produce an object with -g and then strip it, perfectly fine.   By the way there exists other software in OpenIndiana like 'firefox' that is also stripped.